src/explorer/editors/FileEditor.ts

```diff
@@ -36,14 +31,10 @@ export class FileEditor extends BaseEditor<FileTreeItem> {
         return 0;
     }
 
-    public async updateData(node: FileTreeItem): Promise<string> {
-        if (!vscode.window.activeTextEditor) {
-            throw new Error('Cannot update file after it has been closed.');
-        }
-        const localFile: Readable = fs.createReadStream(vscode.window.activeTextEditor.document.uri.fsPath);
-        let etag: string = nonNullValue(this._etags.get(node.fullId), 'etag');
-        etag = await putFile(node.root.client, localFile, node.path, etag);
-        this._etags.set(node.fullId, etag);
+    public async updateData(node: FileTreeItem, data: string): Promise<string> {
+        // because the etag can become stale, get the latest before updating

```

Comment:
Discussed offline. Need to be more careful with etag because we don't want to overwrite changes that might've already been made in the cloud